handler/handler_test.go

```diff
@@ -65,266 +64,279 @@ func TestHandling(t *testing.T) {
 	}
 	h := Create(scheduler, db)
 
-	db.Clean()
 	defer db.Close()
 
-	Convey("writeJSON", t, func() {
-		Convey("Should respond with a JSON and the appropriate status code", func() {
-			var wr = httptest.NewRecorder()
-
-			writeJSON(200, "foo", wr)
-			contentType := wr.HeaderMap["Content-Type"][0]
-			So(contentType, ShouldEqual, "application/json; charset=UTF-8")
-			So(wr.Code, ShouldEqual, http.StatusOK)
-		})
-	})
+	Convey("Routes", t, func() {
+		db.Clean()
 
-	Convey("HandleError", t, func() {
 		wr := httptest.NewRecorder()
-
-		Convey("It should return an error status code", func() {
-			var err = mockError{
-				message: "Error",
-			}
-
-			handleError(err, wr, "A test error")
-
-			So(wr.Code, ShouldEqual, 422)
-			So(strings.TrimSpace(wr.Body.String()), ShouldEqual, "{\"error\":\"Error\",\"message\":\"A test error\"}")
-		})
-	})
-
-	Convey("GetTaskInfo", t, func() {
-		wr := httptest.NewRecorder()
-		r, _ := http.NewRequest("GET", "/task/eremetic-task.1234", nil)
 		m := mux.NewRouter()
-		m.HandleFunc("/task/{taskId}", h.GetTaskInfo(&config.Config{}))
-
-		Convey("Not Found", func() {
-			id := "eremetic-task.5678"
-			task := types.EremeticTask{
-				TaskCPUs: 0.2,
-				TaskMem:  0.5,
-				Command:  "test",
-				Image:    "test",
-				Status:   status,
-				ID:       id,
-			}
-			db.PutTask(&task)
-			m.ServeHTTP(wr, r)
 
-			So(wr.Code, ShouldEqual, http.StatusNotFound)
+		Convey("GetTaskInfo", func() {
+			r, _ := http.NewRequest("GET", "/task/eremetic-task.1234", nil)
+			m.HandleFunc("/task/{taskId}", h.GetTaskInfo(&config.Config{}))
+
+			Convey("JSON request", func() {
+				Convey("Not Found", func() {
+					id := "eremetic-task.5678"
```

Comment:
Maybe it would be cleaner to initialise the test state in `GetTaskInfo` and change the requested url for not found? Could also share the setup between html/json that way